### PR TITLE
Remove the queue from the core

### PR DESCRIPTION
**Pull request by annando**
This finalizes the work from PR https://github.com/friendica/friendica/pull/6972

### CODE REVIEW

**Comment by annando** [comment]
Thanks for the hint. I totally forgot about this.

**Comment by MrPetovan** [comment]
Oh, so existing installs will keep an empty table forever, while newer install won't have one at all. Works for me.

File: update.php [hunk 1]
```diff
@@ -346,3 +347,30 @@ function update_1298()
...
+			$cmd = Delivery::POST;
+		} else {
+			$cmd = Delivery::COMMENT;
+		}
+
+		$deliver_options = ['priority' => PRIORITY_MEDIUM, 'dont_fork' => true];
+		Worker::add($deliver_options, 'Delivery', $cmd, $item['id'], $entry['cid']);
+		Logger::info('Added delivery worker', ['command' => $cmd, 'item' => $item['id'], 'contact' => $entry['cid']]);
+	}
+	return Update::SUCCESS;
```
Comment (MrPetovan) [inline update.php]: Please empty the `queue` table at the end of this update process.

**Comment by annando** [comment]
@MrPetovan Are you still missing something? Or do you wait for others to approve this PR as well?

**Comment by MrPetovan** [comment]
I was waiting for the tests to complete, but as usual I forgot about it because I'm not getting a notification when the tests finish.

**Comment by annando** [comment]
Think of this PR: https://github.com/friendica/friendica-addons/pull/841

**Comment by MrPetovan** [comment]
Thanks, I always forget.

**Comment by AlfredSK** [comment]
Hmm. Since I pulled the latest changes including this PR delivery has stopped completely. I wrote a test post 30 minutes ago but it still shows 'notifier task is pending'. Delivery worked after the first "queue PR" https://github.com/friendica/friendica/pull/6972 but not after this one.

The node I'm talking about is mordor.social, a small (low traffic) node . But still, I see the workerqueue growing slowly since pulling this PR. Even jobs with prio 20 aren't processed anymore.

Any ideas?

**Comment by AlfredSK** [comment]
Also, I get **a lot** (10-20 per minute) of ...

`PHP Notice:  Only variables should be passed by reference in /var/www/html/mordor/src/App.php on line 1273`

... related to the router.

**Comment by AlfredSK** [comment]
Looks like no workers are started except when cron is starting some workers. Then around 4 jobs are processed and the workers stop again (not processing the other waiting jobs). Then nothings happens at all until the next time cron is starting the workers again.

I dropped the workerqueue and restarted Apache and PHP but it didn't help.

This is all what happens when cron is starting the workers:

`2019-04-08 19:36:02 worker [INFO]: Load: 3.83/20 - processes: 0/1/220 (0:1, 20:0/162, 30:0/12, 40:0/46) - maximum: 11/20 [] - {"file":"Worker.php","line":721,"function":"tooMuchWorkers","uid":"46a5f9","process_id":20464}
2019-04-08 19:36:02 worker [INFO]: Load: 3.83/20 - processes: 0/1/220 (0:1, 20:0/162, 30:0/12, 40:0/46) - maximum: 11/20 [] - {"file":"Worker.php","line":721,"function":"tooMuchWorkers","uid":"46a5f9","process_id":20464}
2019-04-08 19:36:02 worker [INFO]: Load: 3.83/20 - processes: 0/1/220 (0:1, 20:0/162, 30:0/12, 40:0/46) - maximum: 11/20 [] - {"file":"Worker.php","line":721,"function":"tooMuchWorkers","uid":"46a5f9","process_id":20464}`

**Comment by AlfredSK** [comment]
There's nothing else in the PHP log. Just the line I posted above.

**Comment by annando** [comment]
Then please have a look in the logfile for all entries with "Worker.php" in it. When you follow all entries with a single "uid" (like "46a5f9") - what do you see?

**Comment by AlfredSK** [comment]
```
2019-04-08 19:36:02 worker [INFO]: Load: 3.83/20 - processes: 0/1/220 (0:1, 20:0/162, 30:0/12, 40:0/46) - maximum: 11/20 [] - {"file":"Worker.php","line":721,"function":"tooMuchWorkers","uid":"46a5f9","process_id":20464}
2019-04-08 19:36:02 worker [INFO]: Active workers: 1/11 Fork a new worker. [] - {"file":"Worker.php","line":725,"function":"tooMuchWorkers","uid":"46a5f9","process_id":20464}
2019-04-08 19:36:02 worker [INFO]: Connection usage (system values): 19/128 [] - {"file":"Worker.php","line":529,"function":"maxConnectionsReached","uid":"46a5f9","process_id":20464}
2019-04-08 19:36:02 worker [INFO]: Add cron entries [] - {"file":"Worker.php","line":1065,"function":"runCron","uid":"46a5f9","process_id":20464}
2019-04-08 19:36:02 worker [INFO]: Load: 3.83/20 - processes: 0/1/220 (0:1, 20:0/162, 30:0/12, 40:0/46) - maximum: 11/20 [] - {"file":"Worker.php","line":721,"function":"tooMuchWorkers","uid":"46a5f9","process_id":20464}
2019-04-08 19:36:02 worker [INFO]: Active workers: 1/11 Fork a new worker. [] - {"file":"Worker.php","line":725,"function":"tooMuchWorkers","uid":"46a5f9","process_id":20464}
2019-04-08 19:36:02 worker [INFO]: Load: 3.83/20 - processes: 0/1/220 (0:1, 20:0/162, 30:0/12, 40:0/46) - maximum: 11/20 [] - {"file":"Worker.php","line":721,"function":"tooMuchWorkers","uid":"46a5f9","process_id":20464}
2019-04-08 19:36:02 worker [INFO]: Active workers: 1/11 Fork a new worker. [] - {"file":"Worker.php","line":725,"function":"tooMuchWorkers","uid":"46a5f9","process_id":20464}
2019-04-08 19:36:02 worker [INFO]: No running worker found with priority 20 - assigning it. {"priority":20} - {"file":"Worker.php","line":847,"function":"nextPriority","uid":"46a5f9","process_id":20464}
2019-04-08 19:36:02 worker [INFO]: Found: {"priority":20,"id":[501449,501453,501454,501459]} - {"file":"Worker.php","line":806,"function":"nextProcess","uid":"46a5f9","process_id":20464}
2019-04-08 19:36:02 worker [INFO]: Connection usage (system values): 19/128 [] - {"file":"Worker.php","line":529,"function":"maxConnectionsReached","uid":"46a5f9","process_id":20464}
2019-04-08 19:36:02 worker [INFO]: Process start. {"priority":20,"id":501449,"worker_id":"502d94a","worker_cmd":"Delivery"} - {"file":"Worker.php","line":384,"function":"execFunction","uid":"46a5f9","process_id":20464}
2019-04-08 19:36:04 worker [INFO]: Performance: {"state":1,"count":0,"stat":0.01,"write":0,"lock":0,"total":0.03,"rest":0.14,"exec":1.7} - {"file":"Worker.php","line":426,"function":"execFunction","uid":"46a5f9","process_id":20464}
2019-04-08 19:36:04 worker [INFO]: Process done. {"priority":20,"id":501449,"duration":1.702,"worker_id":"502d94a","worker_cmd":"Delivery"} - {"file":"Worker.php","line":445,"function":"execFunction","uid":"46a5f9","process_id":20464}
2019-04-08 19:36:04 worker [INFO]: ID 501449: Delivery {"action":"profiling","database_read":0.012,"database_write":0.306,"cache_read":0,"cache_write":0,"network_io":1.2,"file_io":0,"other_io":0.19,"total":1.7} - {"file":"Worker.php","line":447,"function":"execFunction","uid":"46a5f9","process_id":20464}
2019-04-08 19:36:04 worker [INFO]: No running worker found with priority 30 - assigning it. {"priority":30} - {"file":"Worker.php","line":847,"function":"nextPriority","uid":"46a5f9","process_id":20464}
2019-04-08 19:36:04 worker [INFO]: Found: {"priority":30,"id":[501311]} - {"file":"Worker.php","line":806,"function":"nextProcess","uid":"46a5f9","process_id":20464}
2019-04-08 19:36:04 worker [INFO]: Connection usage (system values): 18/128 [] - {"file":"Worker.php","line":529,"function":"maxConnectionsReached","uid":"46a5f9","process_id":20464}
2019-04-08 19:36:04 worker [INFO]: Process start. {"priority":20,"id":501453,"worker_id":"c04d15c","worker_cmd":"Delivery"} - {"file":"Worker.php","line":384,"function":"execFunction","uid":"46a5f9","process_id":20464}
2019-04-08 19:36:04 worker [INFO]: Performance: {"state":3,"count":0,"stat":0,"write":0.01,"lock":0,"total":0.01,"rest":0.01,"exec":0.31} - {"file":"Worker.php","line":426,"function":"execFunction","uid":"46a5f9","process_id":20464}
2019-04-08 19:36:04 worker [INFO]: Process done. {"priority":20,"id":501453,"duration":0.306,"worker_id":"c04d15c","worker_cmd":"Delivery"} - {"file":"Worker.php","line":445,"function":"execFunction","uid":"46a5f9","process_id":20464}
2019-04-08 19:36:04 worker [INFO]: ID 501453: Delivery {"action":"profiling","database_read":0.009,"database_write":0.002,"cache_read":0,"cache_write":0,"network_io":0.26,"file_io":0,"other_io":0.03,"total":0.31} - {"file":"Worker.php","line":447,"function":"execFunction","uid":"46a5f9","process_id":20464}
2019-04-08 19:36:04 worker [INFO]: Connection usage (system values): 17/128 [] - {"file":"Worker.php","line":529,"function":"maxConnectionsReached","uid":"46a5f9","process_id":20464}
2019-04-08 19:36:04 worker [INFO]: Process start. {"priority":20,"id":501454,"worker_id":"fc37ea3","worker_cmd":"Delivery"} - {"file":"Worker.php","line":384,"function":"execFunction","uid":"46a5f9","process_id":20464}
2019-04-08 19:36:06 worker [INFO]: Performance: {"state":4,"count":0,"stat":0,"write":0,"lock":0,"total":0,"rest":0.01,"exec":2.34} - {"file":"Worker.php","line":426,"function":"execFunction","uid":"46a5f9","process_id":20464}
2019-04-08 19:36:06 worker [INFO]: Process done. {"priority":20,"id":501454,"duration":2.34,"worker_id":"fc37ea3","worker_cmd":"Delivery"} - {"file":"Worker.php","line":445,"function":"execFunction","uid":"46a5f9","process_id":20464}
2019-04-08 19:36:06 worker [INFO]: ID 501454: Delivery {"action":"profiling","database_read":0.01,"database_write":0.002,"cache_read":0,"cache_write":0,"network_io":2.3,"file_io":0,"other_io":0.03,"total":2.34} - {"file":"Worker.php","line":447,"function":"execFunction","uid":"46a5f9","process_id":20464}
2019-04-08 19:36:06 worker [INFO]: Connection usage (system values): 19/128 [] - {"file":"Worker.php","line":529,"function":"maxConnectionsReached","uid":"46a5f9","process_id":20464}
2019-04-08 19:36:06 worker [INFO]: Process start. {"priority":20,"id":501459,"worker_id":"0ef860f","worker_cmd":"Delivery"} - {"file":"Worker.php","line":384,"function":"execFunction","uid":"46a5f9","process_id":20464}
2019-04-08 19:36:07 worker [INFO]: Performance: {"state":4,"count":0,"stat":0,"write":0.01,"lock":0,"total":0.01,"rest":0.01,"exec":0.62} - {"file":"Worker.php","line":426,"function":"execFunction","uid":"46a5f9","process_id":20464}
2019-04-08 19:36:07 worker [INFO]: Process done. {"priority":20,"id":501459,"duration":0.617,"worker_id":"0ef860f","worker_cmd":"Delivery"} - {"file":"Worker.php","line":445,"function":"execFunction","uid":"46a5f9","process_id":20464}
2019-04-08 19:36:07 worker [INFO]: ID 501459: Delivery {"action":"profiling","database_read":0.01,"database_write":0.002,"cache_read":0,"cache_write":0,"network_io":0.57,"file_io":0,"other_io":0.03,"total":0.62} - {"file":"Worker.php","line":447,"function":"execFunction","uid":"46a5f9","process_id":20464}
2019-04-08 19:36:07 worker [INFO]: Process lifetime reached, respawning. [] - {"file":"Worker.php","line":154,"function":"processQueue","uid":"46a5f9","process_id":20464}
```

**Comment by annando** [comment]
Concerning the logs, the system is doing something. You can see it in the log entries with "Process done". You could run something like this:
```tail -f logfile | grep done.*Worker.php``` 
Then you should see, how fast and often it appears. When your system is sane, then this should appear very often.

**Comment by AlfredSK** [comment]
There are exactly 4 worker jobs done each time cron is running the worker. The cronjob is scheduled every 4 minutes. That makes exactly 60 jobs per hour or 1 jpm. That's it.

So, delivery works but it's so slow that each post takes multiple hours until it is delivered to all recipients.

**Comment by AlfredSK** [comment]
The 4 jobs are only processed as long as I have worker_fetch_limit set to 4. When I comment this config line out only 1 job is done on each cron round. That makes 0.25 jpm. :-/

**Comment by annando** [comment]
Does this behaviour change when you are running the daemon? Just enter ```bin/daemon.php start``` for this.

It seems as if the spawning doesn't work. And if only a few jobs are called, but after a few seconds the system tells "Process lifetime reached", then there is something strange.

**Comment by annando** [comment]
@AlfredSK I guess I found the origin of the problem, here is the PR for this: https://github.com/friendica/friendica/pull/6993

**Comment by AlfredSK** [comment]
Thanks @annando . It's working better now. Still slow but it works.
There is one thing I don't understand. When I run the worker manually on a master branch node I get no output. But when I do it with mordor.social (on develop but on the same machine) I get this:

```
cd /var/www/mordor ; sudo -u www-data /usr/bin/php bin/worker.php
sh: 1: usr/bin/php: not found
sh: 1: usr/bin/php: not found
sh: 1: usr/bin/php: not found
sh: 1: usr/bin/php: not found
sh: 1: usr/bin/php: not found
sh: 1: usr/bin/php: not found
sh: 1: usr/bin/php: not found
sh: 1: usr/bin/php: not found
sh: 1: usr/bin/php: not found
sh: 1: usr/bin/php: not found
sh: 1: usr/bin/php: not found
sh: 1: usr/bin/php: not found
sh: 1: usr/bin/php: not found
sh: 1: usr/bin/php: not found
sh: 1: usr/bin/php: not found
sh: 1: usr/bin/php: not found
sh: 1: usr/bin/php: not found
sh: 1: usr/bin/php: not found
sh: 1: usr/bin/php: not found
sh: 1: usr/bin/php: not found
sh: 1: usr/bin/php: not found
sh: 1: usr/bin/php: not found
sh: 1: usr/bin/php: not found
sh: 1: usr/bin/php: not found
sh: 1: usr/bin/php: not found
sh: 1: usr/bin/php: not found
```

Is this related or something completely different?

**Comment by annando** [comment]
Looks like the config for "php_path" in "local.config.php" directs to some wrong path.

**Comment by tobiasd** [comment]
> usr/bin/php

A missing "/" at the beginning of the command?

**Comment by AlfredSK** [comment]
Geez! :open_mouth:  Sorry. I checked the config like 20 times after converting it from INI to the latest style. But I didn't notice the missing slash. :man_facepalming:

**Comment by annando** [comment]
@AlfredSK How fast is it now?

**Comment by AlfredSK** [comment]
Lightning fast.

**Pull request merged**